tests/test_diffraction_fft.py

```diff
@@ -0,0 +1,477 @@
+    expected_img = np.array(
+        [
+            [1, 0, 0, 0, 0, 0],
+            [1, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 0],
+        ]
+    )
+    np.testing.assert_array_equal(img_a, expected_img)
+
+
+def test_add_image_works_for_overlay_overlapping_img_rb():
+    img_a = np.zeros((5, 6), dtype=int)
+    img_b = np.ones((3, 3), dtype=int)
+    diffraction_fft.add_image(img_a, img_b, 4, 4)
+    expected_img = np.array(
+        [
+            [0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 0, 0],
+            [0, 0, 0, 0, 1, 1],
+        ]
+    )
+    np.testing.assert_array_equal(img_a, expected_img)
+
+
+def test_add_image_works_for_covering_overlay():
+    img_a = np.zeros((5, 6), dtype=int)
+    img_b = np.ones((7, 7), dtype=int)
+    diffraction_fft.add_image(img_a, img_b, -1, 0)
+    expected_img = np.ones((5, 6), dtype=int)
+    np.testing.assert_array_equal(img_a, expected_img)
+
+
+def test_saturated_clusters_works_for_no_saturated_pixels():
+    img = np.zeros((5, 6), dtype=int)
+    regions = diffraction_fft.saturated_clusters(img, 0.5)
+    assert not regions
+
+
+def test_saturated_clusters_works_for_1_saturated_pixel():
+    img = np.zeros((5, 6), dtype=int)
+    img[2, 3] = 1.0
+    regions = diffraction_fft.saturated_clusters(img, 0.5)
+    assert regions == [(slice(2, 3), slice(3, 4))]
+
+
+def test_saturated_clusters_finds_saturated_clusters():
+    img = np.array(
+        [
+            [0, 1, 0, 0, 0, 0],
+            [0, 0, 0, 0, 1, 1],
+            [1, 0, 1, 0, 0, 0],
+            [0, 1, 0, 0, 0, 1],
+            [1, 0, 1, 0, 0, 1],
+        ]
+    )
+    regions = diffraction_fft.saturated_clusters(img, 0.5)
+    expected_regions = [
+        (slice(0, 1), slice(1, 2)),
+        (slice(1, 2), slice(4, 6)),
+        (slice(2, 5), slice(0, 3)),
+        (slice(3, 5), slice(5, 6)),
+    ]
+    assert sorted(regions) == sorted(
+        expected_regions
+    ), f"{sorted(regions)} != {sorted(expected_regions)}"
+
+
+def test_fft_diffraction_is_similar_to_raytracing_for_0_exptime():
+    """Check, that raytracing diffraction and FFT diffraction are "similar"."""
+    config = create_test_config(XSIZE, YSIZE, STAMP_SIZE, exptime=0.0)
+    # To compare with the raytracing method, instead use:
+    # config = create_test_config(XSIZE, YSIZE, STAMP_SIZE, exptime=0.0, mode=Mode.RAYTRACING)
+
+    with assert_no_error_logs() as logger:
+        image = galsim.config.BuildImage(config, logger=logger)
+    # To save the image produced here, use:
+    # galsim.fits.write(image, file_name="/tmp/spikes.fits")
+    brightness = image.array
+    # Center of star:
+    [c_x, c_y] = center_of_brighness(brightness)
+    # 2 Pixel tolerance:
+    np.testing.assert_allclose(
+        np.array([c_x, c_y]), np.array([1923.0, 3161.0]), atol=2.0, rtol=0.0
```

Comment:
Where did (1923, 3161) come from?  I was expecting to see one image built with raytracing and the other with fft.
I'd also expect them to be closer than 2 pixels apart.  The center of brightness should be much less than 1 pixel different I would think.  (Especially for non-rotation, but even with rotation.)

Reply:
All those magic numbers (here and below) actually do come from a run with raytracing.
I designed the tests as suggested by @cwwalter such that the reference values are generated once using rytracing mode and do not need to be recomputed during CI testing.

(thats also why the comment 
```
    # To compare with the raytracing method, instead use:
    # config = create_test_config(XSIZE, YSIZE, STAMP_SIZE, exptime=0.0, mode=Mode.RAYTRACING)
```
is there).
But here, I guess it would not add that much additional compute time to also integrate the raytracing mode runs.
What do you think?

Reply:
If the ray tracing is too time consuming to compute in regular CI runs, then you should still have code in there to generate it, but have it not run normally.
E.g.
```
file_name = 'data/raytrace_diffraction_values.npz'
if not os.path.isfile(file_name):
    # ... Run ray tracing code to generate values.
    np.savez(file_name, c_x=c_x, c_y=c_y, ...)

raytrace_data = np.load(file_name)

# ... Use raytrace_data['c_x'], raytrace_data['c_y'], to compare FFT version.
```
Then you run this locally and commit the data file to the repo.  If something changes where we need to recompute things, we delete the file and run again.